pandas/tests/sparse/frame/test_frame.py

```diff
@@ -156,6 +156,15 @@ def test_constructor_ndarray(self, float_frame):
             SparseDataFrame(float_frame.values,
                             columns=float_frame.columns[:-1])
 
+    def test_construct_unmatched_sparsedataframe(self, float_frame):
+        # GH25484
+        dates = pd.bdate_range('1/1/2011', periods=10)
+        sp = SparseDataFrame(float_frame.values, index=dates)
+        N = len(sp)
+        msg = 'Length of values does not match length of index'
+        with pytest.raises(ValueError, match=msg):
+            sp['foo'] = np.random.randn(N-1)
+
```

Comment:
may not need an additional test (https://github.com/pandas-dev/pandas/pull/25503#issuecomment-468906150)

the code sample in #25484 replicates the setup of the existing test. so maybe able to alter existing test to hit both changes you've made to `pandas\core\sparse\frame.py in _sanitize_column`

Reply:
ahh, i see! thanks! @simonjayhawkins